frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -72,11 +73,12 @@ export function NotebookDataPicker({
 
   if (isEmbedding) {
     return (
-      <LegacyDataPicker
+      <EmbeddingDataPicker
         query={query}
         stageIndex={stageIndex}
         table={table}
         placeholder={placeholder}
+        canChangeDatabase={canChangeDatabase}
```

Comment:
I just realized I just passed this prop here, so I go back and tested `1.52.10` and found out that we don't prevent join across DB in the multi-stage data picker.

<img width="1350" alt="Screenshot 2025-02-11 at 5 59 23 PM" src="https://github.com/user-attachments/assets/57c8c9f0-7d5c-4a73-9ff1-498641b6ad78" />

I added a task to [milestone 1](https://linear.app/metabase/issue/EMB-120/milestone-1-the-flattened-list-fallback-to-multi-stage-when-data), and this will be addressed in the next PR, just to keep this PR from getting any bigger.